torch/distributed/device_mesh.py

```diff
@@ -484,70 +530,52 @@ def _init_process_groups(self):
             # https://github.com/pytorch/pytorch/issues/93173#issuecomment-1907095208
             dim_group_infos: List[Tuple[str, List[int], str]] = []
 
-            if self.mesh.ndim == 1 and self.mesh.numel() == get_world_size():
-                # Append the default pg to the first dim groups only if the default pg is compatible with `self.device_type`.
-                # Otherwise, create new pg.
-                default_group = _get_default_group()
-                ranks = list(range(get_world_size()))
-                dim_group = (
-                    new_group(backend="cpu:gloo,cuda:nccl", ranks=ranks)
-                    if torch.cuda.is_available()
-                    and get_backend(default_group) == "gloo"
-                    else default_group
```

Comment:
I removed this `if` branch (the special case we used to take care of per https://github.com/pytorch/pytorch/pull/132709)
@wz337